### PR TITLE
fix #204 | get this.options.eslint without error

### LINKED ISSUE #204: loaderContext.options were removed from webpack 4.0.0-alpha.0
**eslint-loader** fails with error when running with webpack 4.0.0-alpha.0 https://github.com/webpack/webpack/issues/6064

output:
```
Module build failed: TypeError: Cannot read property 'eslint' of undefined
    at Object.module.exports (/Users/svyatoslav/Dev/wix-vod/wix-vod/node_modules/eslint-loader/index.js:148:18)
webpack: Failed to compile.
```

### PR DESCRIPTION
**Pull request by wonism**
#204

### CODE REVIEW

**Comment by wonism** [comment]
There's an error when user uses node version that is under 6.
So, I suggest to remove `4` on `node_js` in `.travis.yml` file.

```
$ yarn
yarn install v1.3.2
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@1.1.2: The platform "linux" is incompatible with this module.
info "fsevents@1.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
error webpack@4.0.0-beta.2: The engine "node" is incompatible with this module. Expected version ">=6.11.5".
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by wonism** [comment]
There's an error when users use node version that is under 6.
I suggest to remove this in `.travis.yml`.

```
$ yarn
yarn install v1.3.2
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@1.1.2: The platform "linux" is incompatible with this module.
info "fsevents@1.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
error webpack@4.0.0-beta.2: The engine "node" is incompatible with this module. Expected version ">=6.11.5".
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by MoOx** [comment]
Look nice.
What about removing support for older version to simplify the tests? See https://github.com/webpack-contrib/eslint-loader/pull/210#issuecomment-365179162

**Comment by wonism** [comment]
@MoOx
You mean stopping to support webpack v1?
If not, I will revert the commit.

**Comment by stnwk** [comment]
Thanks @wonism for this!

**Comment by theskillwithin** [comment]
Thanks can't wait till this goes out! =]

**Comment by rickbeerendonk** [comment]
fix #215

**Comment by stnwk** [comment]
@wonism @MoOx Now that webpack 4.0.0 was released stable https://github.com/webpack/webpack/releases/tag/v4.0.0 - any chance we can get this out soon?

**Comment by rickbeerendonk** [comment]
The Webpack version change in .travis.yml needs to be corrected before merging (now that Webpack 4.0.0 is released).

File: package.json [hunk 1]
```diff
@@ -23,14 +23,14 @@
     "loader-utils": "^1.0.2",
     "object-assign": "^4.0.1",
     "object-hash": "^1.1.4",
-    "rimraf": "^2.6.1"
+    "rimraf": "^2.6.1",
+    "webpack": "^4.0.0"
```
Comment (MoOx) [inline package.json]: Should not be added as a dep, just devDep

Comment (wonism) [reply package.json]: Oh, I forgot the flag when I install the webpack. I fixed it :)

**Comment by MoOx** [comment]
Thanks for the work

**Comment by MoOx** [comment]
Released in 2.0.0

**Pull request merged**